circadian/models.py

```diff
@@ -1,878 +1,950 @@
+
+# %% ../nbs/00_models.ipynb 33
+@patch_to(CircadianModel)
+def get_parameters_array(self)-> np.array:
+    "Returns the parameters for the model as a numpy array"
+    parameter_array = np.zeros(len(self.parameters))
+    for idx, value in enumerate(self.parameters.values()):
+        parameter_array[idx] = value
+    return parameter_array
+
+# %% ../nbs/00_models.ipynb 35
+@patch_to(CircadianModel)
+def phase(self,
+          trajectory: DynamicalTrajectory=None, # trajectory to calculate the phase for. If None, the phase is calculated for the current trajectory 
+          time: float=None # timepoint to calculate the phase at. If None, the phase is calculated for the entire trajectory
+          ) -> float:
+    "Calculates the phase of the model at a given timepoint"
+    raise NotImplementedError("phase is not implemented for this model")
+
+# %% ../nbs/00_models.ipynb 36
+@patch_to(CircadianModel)
+def amplitude(self,
+              trajectory: DynamicalTrajectory=None, # trajectory to calculate the amplitude for. If None, the amplitude is calculated for the current trajectory 
+              time: float=None, # timepoint to calculate the amplitude at. If None, the amplitude is calculated for the entire trajectory
+              ) -> float:
+    "Calculates the amplitude of the model at a given timepoint"
+    raise NotImplementedError("amplitude is not implemented for this model")
+
+# %% ../nbs/00_models.ipynb 37
+@patch_to(CircadianModel)
+def cbt(self,
+        trajectory: DynamicalTrajectory=None, # trajectory to calculate the cbt for. If None, the cbt is calculated for the current trajectory
+        ) -> np.ndarray: # array of times when the cbt occurs
+    "Finds the core body temperature minumum markers along a trajectory"
+    raise NotImplementedError("cbt is not implemented for this model")
+
+# %% ../nbs/00_models.ipynb 38
+@patch_to(CircadianModel)
+def dlmos(self,
+          trajectory: DynamicalTrajectory=None, # trajectory to calculate the dlmos for. If None, the dlmos are calculated for the current trajectory
+          ) -> np.ndarray: # array of times when dlmo occurs 
+    "Finds the Dim Light Melatonin Onset (DLMO) markers along a trajectory"
+    raise NotImplementedError("dlmo is not implemented for this model")
+
+# %% ../nbs/00_models.ipynb 39
+@patch_to(CircadianModel)
+def equilibrate(self,
+                time: np.ndarray, # time points for integration. Time difference between each consecutive pair of values determines step size of the solver
+                input: np.ndarray, # model input (such as light or wake) for each time point
+                num_loops: int=10 # number of times to loop the input
+                ) -> np.ndarray: # final state of the model
+    "Equilibrate the model by looping the given light_estimate. Assumes the schedule repeats periodically after it ends"
+    # input checking
+    _time_input_checking(time)
+    _model_input_checking(input, self._num_inputs, time)
```

Comment:
Let's add a warning for when a input does not equilibrate after 10 loops by checking the last two CBTs or DLMOs (from the loop) and verifying 

- LOOP 1
- LOOP 2
...
- LOOP N-1
- LOOP N

Assert the DLMO at the end of LOOP N-1 is within some tolerance of LOOP N
If not, warn user.

Reply:
Did exactly this at the end of the `equilibrate()` function